src/renderer/Drawable.ts

```diff
@@ -18,30 +24,44 @@ const determinant = (a, b, c) => {
 // TODO: store renderer-specific data on the sprite and have *it* set a
 // "transform changed" flag.
 class SpriteTransformDiff {
-  constructor(sprite) {
+  private _sprite: Sprite | Stage;
+  private _unset: boolean;
+
+  private _lastX!: Sprite["x"] | undefined;
+  private _lastY!: Sprite["y"] | undefined;
+  private _lastRotation!: Sprite["direction"] | undefined;
+  private _lastRotationStyle!: Sprite["rotationStyle"] | undefined;
+  private _lastSize!: Sprite["size"] | undefined;
+  private _lastCostume!: Sprite["costume"];
+  private _lastCostumeLoaded!: boolean;
```

Comment:
OK, this block confuses me :) I thought I understood the use of `!` for indicating that even though a variable's type implies it could be null/undefined, it certainly isn't at this point... but here we're explicitly allowing for `undefined` at the same time as we define fields. I'd really appreciate some details on the usage here.

Reply:
I think I added them then realized the properties could be undefined. I've removed them now.